sys/Makefile.include

```diff
@@ -35,26 +35,7 @@ ifneq (,$(filter vfs,$(USEMODULE)))
   USEMODULE_INCLUDES += $(RIOTBASE)/sys/posix/include
 endif
 
-ifneq (,$(filter cpp11-compat,$(USEMODULE)))
-  USEMODULE_INCLUDES += $(RIOTBASE)/sys/cpp11-compat/include
-  # make sure cppsupport.o is linked explicitly because __dso_handle is not
-  # found if it is hidden away inside a static object.
-  export UNDEF += $(BINDIR)/cpp11-compat/cppsupport.o
-endif
-
-ifneq (,$(filter embunit,$(USEMODULE)))
-  ifeq ($(OUTPUT),XML)
-    CFLAGS += -DOUTPUT=OUTPUT_XML
-  else ifeq ($(OUTPUT),TEXT)
-    CFLAGS += -DOUTPUT=OUTPUT_TEXT
-  else ifeq ($(OUTPUT),COMPILER)
-    CFLAGS += -DOUTPUT=OUTPUT_COMPILER
-  else ifeq ($(OUTPUT),COLORTEXT)
-    CFLAGS += -DOUTPUT=OUTPUT_COLORTEXT
-  else ifeq ($(OUTPUT),COLOR)
-    CFLAGS += -DOUTPUT=OUTPUT_COLOR
-  endif
-endif
+-include $(patsubst %,$(RIOTBASE)/sys/%/Makefile.include,$(USEMODULE))
```

Comment:
As discussed IRL, I am not sure I want to go in this direction for the moment.

It goes in a direction were module names must be a directory name at the root of `sys`, and we know it does not work or apply for everything.

I would find it weird that for example the posix module gets put in a flat hierarchy:

```
ls sys/posix/
doc.txt  include  inet.c  Makefile  pthread  semaphore  sockets  time
```
as 
```
sys/posix
sys/posix_pthread
sys/posix_semaphore
...
```

And would not want to rename the directory inside to `posix/posix_pthread`.

It pushes to a new hierarchy, without giving the flexibility currently required for what we want to do.

----

However this IRL long intense discussions ended up going more for `posix/pthread` module names to have a deterministic way of finding directly where a module is implemented if you know all the module trees.